cmd/snap/cmd_routine_file_access_test.go

```diff
@@ -0,0 +1,184 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package main_test
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+	"os/user"
+	"path/filepath"
+	"strings"
+
+	. "gopkg.in/check.v1"
+
+	"github.com/snapcore/snapd/client"
+	snap "github.com/snapcore/snapd/cmd/snap"
+)
+
+type SnapRoutineFileAccessSuite struct {
+	BaseSnapSuite
+
+	fakeHome string
+}
+
+var _ = Suite(&SnapRoutineFileAccessSuite{})
+
+func (s *SnapRoutineFileAccessSuite) SetUpTest(c *C) {
+	s.BaseSnapSuite.SetUpTest(c)
+
+	s.fakeHome = c.MkDir()
+	u, err := user.Current()
+	c.Assert(err, IsNil)
+	s.AddCleanup(snap.MockUserCurrent(func() (*user.User, error) {
+		return &user.User{Uid: u.Uid, HomeDir: s.fakeHome}, nil
+	}))
+}
+
+func (s *SnapRoutineFileAccessSuite) setUpClient(c *C, isClassic, hasHome, hasRemovableMedia bool) {
+	s.RedirectClientToTestServer(func(w http.ResponseWriter, r *http.Request) {
+		switch r.URL.Path {
+		case "/v2/snaps/hello":
+			c.Check(r.Method, Equals, "GET")
+			response := mockInfoJSONNoLicense
```

Comment:
given this is kind of far away (this package has so many files),  probably worth a comment before

// snap hello at revision 100